### PR TITLE
add Get A Shopcart Item API

### PR DESCRIPTION
**Pull request by momo100110**
add Get A Shopcart Item API and test cases 
image

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -138,33 +165,28 @@ def add_shopcart_item(shopcart_id):
...
-    #     return internal_server_error(e)
-    shopcart = Shopcart.get_by_id(shopcart_id)
+    This endpoint returns just a item
+    """
+    app.logger.info(
+        "Request to retrieve Item %s for Shopcart id: %s", item_id, shopcart_id)
 
-    if not shopcart:
+    # See if the item exists and abort if it doesn't
+    item = Item.get_by_id(item_id)
```
[the remark this answers is not in the document]
Comment (wl2484) [reply service/routes.py:181]: There's another advantage of retrieving the item from shopart: this way we can validate whether the `shopcart_id` given in API path exists in DB or not in the same time. If the given `shopcart_id` does not exist, we should produce a 404 Not Found error.

[the remark this answers is not in the document]
Comment (wl2484) [reply service/routes.py:181]: I'm fine if you choose to use `Item.get_by_id()` 🙂 However, please make sure that the API would give `404 Not Found` for a non-existent `shopcart_id` in API path. Thanks!

File: tests/test_routes.py [hunk 2]
```diff
@@ -112,6 +112,39 @@ def test_get_shopcart_not_found(self):
...
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        data = response.get_json()
+        logging.debug(data)
+        self.assertEqual(data["shopcart_id"], test_shopcart.id)
+        self.assertEqual(data["quantity"],item.quantity)
+        self.assertEqual(data["price"],item.price)
+
```
Comment (wl2484) [inline tests/test_routes.py:142]: we would need one more test case for getting an item from a non-existent `shopcart_id`

File: tests/test_routes.py [hunk 3]
```diff
@@ -112,6 +112,39 @@ def test_get_shopcart_not_found(self):
...
+        response = self.client.get(
+            f"{BASE_URL}/{test_shopcart.id}/items/{item_id}",
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        data = response.get_json()
+        logging.debug(data)
+        self.assertEqual(data["shopcart_id"], test_shopcart.id)
+        self.assertEqual(data["quantity"],item.quantity)
```
Comment (wl2484) [inline tests/test_routes.py:140]: should we also check the value of `item.name`?

**Review by wl2484 — CHANGES_REQUESTED** [review]

File: tests/test_routes.py (hunk 3 above)
Comment (momo100110) [reply tests/test_routes.py:140]: Yes, thank you for reminding this.

File: service/routes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (momo100110) [reply service/routes.py:190]: Now the API would give 404 Not Found for a non-existent shopcart_id in API path. I'll try your approach after all the APIs are implemented. Thanks for advice!

**Review by wl2484 — APPROVED** [review]
LGTM

**Pull request merged**